sites/target.py

```diff
@@ -0,0 +1,178 @@
+from selenium import webdriver
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait as wait
+from webdriver_manager.chrome import ChromeDriverManager
+from chromedriver_py import binary_path as driver_path
+from utils import random_delay, send_webhook, change_driver
+import settings, time
+
+class Target:
+    def __init__(self, task_id, status_signal, image_signal, product, profile, proxy, monitor_delay, error_delay):
+        self.task_id, self.status_signal, self.image_signal, self.product, self.profile, self.monitor_delay, self.error_delay = task_id, status_signal, image_signal, product, profile, float(
+            monitor_delay), float(error_delay)
+
+        starting_msg = "Starting Target"
+        self.browser = self.init_driver(driver_path)
+        self.product_image = None
+
+        if settings.dont_buy:
+            starting_msg = "Starting Target in dev mode; will not actually checkout"
+
+        self.status_signal.emit(self.create_msg(starting_msg, "normal"))
+        self.status_signal.emit(self.create_msg("Logging In..", "normal"))
+        self.login()
+        self.monitor()
+        self.atc()
+        self.checkout()
+        self.submit_billing()
+
+        if not settings.dont_buy:
+            self.submit_order()
+        else:
+            self.status_signal.emit(self.create_msg("Mock Order Placed", "success"))
+            send_webhook("OP", "Target", self.profile["profile_name"], self.task_id, self.product_image)
+
+    def init_driver(self):
+        driver_manager = ChromeDriverManager()
+        driver_manager.install()
+        change_driver(driver_path)
+        browser = webdriver.Chrome(driver_path)
+
+        browser.execute_cdp_cmd("Page.addScriptToEvaluateOnNewDocument", {
+            "source": """
+                  Object.defineProperty(navigator, 'webdriver', {
+                   get: () => undefined
+                  })
+                """
+        })
+
+        return browser
+
+    def login(self):
+        self.browser.get("https://www.target.com")
+
+        self.browser.find_element_by_id("account").click()
+
+        wait(self.browser, 3).until(EC.element_to_be_clickable((By.ID, "accountNav-signIn"))).click()
+
+        my_username = "username"
```

Comment:
This needs to be configured from the settings page before I can allow it to be merged into master.